docs/troubleshooting-guide/sql-logic.md

```diff
@@ -0,0 +1,261 @@
+| A `INNER JOIN` B | No duplicate rows. | No duplicate rows. | You will get duplicated rows from A or B. |
+| A `LEFT JOIN` B | No duplicate rows. | You may get duplicated rows because of matches from table B. | You will get duplicated rows from A or B. |
+| B `LEFT JOIN` A | No duplicate rows. | You will get duplicated rows because of matches from table B. | You will get duplicated rows from A or B. |
+| A `OUTER JOIN` B | No duplicate rows. | You may get duplicated rows because of matches from table B. | You will get duplicated rows from A or B. |
+| A `FULL JOIN` B | No duplicate rows. | You will get duplicated rows because of matches from table B. | You will get duplicated rows from A or B. |
+
+
+**Explanation**
+
+Rows can get duplicated by accident when data gets refreshed in upstream systems or ETL jobs. 
+
+Some tables have rows that look like duplicates at a glance. This is common with tables that track state changes (e.g. an order status table that adds a row every time the status changes) may have have rows that look exactly the same, except for the timestamp of the row. It can be difficult to detect if you have tables with a lot of columns, so be sure to run through Step 2 above or ask your database admin if you're unsure.
+
+If you’ve written your joins assuming a one-to-one relationship for tables that actually have a one-to-many or many-to-many relationship, you'll get duplicated rows for _each_ match in the "many" table.
+
+**Further reading**
+
+- [What is a schema?][schema-def]
+- [SQL join types][types-of-joins]
+
+#### My result has missing rows
+
+1. Check if your source tables or query results have missing rows. 
+2. Check the [table below](#how-joins-filter-out-unmatched-rows) to see if you are missing rows because of your join type.
+3. Check your join conditions in the `ON` clause. For example:
+    ```sql
+    # The join condition below will remove:
+    # All rows from table A where key_a = "foo"
+    # All rows from table B where key_b = "foo"
+
+    SELECT *
+    FROM table_a
+    JOIN table_b
+    ON key_a = key_b
+    AND key_b <> "foo"
+    ```
+4. Check if your `WHERE` clause is interacting with your `JOIN` clause. For example:
+    ```sql
+    # The WHERE clause below will remove:
+    # All rows from table A where key_a = "foo"
+    # All rows from table B where key_b = "foo"
+
+    SELECT *
+    FROM table_a
+    JOIN table_b
+    WHERE key_b <> "foo"
+    ```
+5. If you want to *add* rows to your query result to fill in data that is empty, zero, or `NULL`, you need to start your JOINs with a table or column that has all the rows you want. Ask your database admin if there’s a table you can use for this, or can create a temporary column using `GENERATE_SERIES`.
+    ```sql
+    # This result has one row for every day with at least one transaction.
+    SELECT 
+    o.date
+    , SUM(p.price) AS total_sales
+    FROM orders o
+    LEFT JOIN products p
+    ON o.product_id = p.id
+    GROUP BY 1
+    ;
```

Comment:
```suggestion
    SELECT 
      o.created_at, 
    SUM(p.price) AS total_sales 
  FROM 
    orders o 
  LEFT JOIN products p ON o.product_id = p.id 
  GROUP BY 
    o.created_at
```

If you include code that uses the Sample Database, might as well make sure it runs.